### DISCUSSION TITLE
[D2G] GUI für YAML-Generierung

### DISCUSSION THREAD

**Discussion by AKirsch1** [Programmiermethoden/Deploy-to-Grading#148] [Ideas]
Basierend auf folgendem Kommentar könnten wir eine GUI zum Erstellen der yaml-Dateien basteln. Webbasiert, da das wahrscheinlich der einfachste Weg ist, die an Studis zu verteilen. Könnte man z.B. über das ILIAS HTML Lernmodul hosten?

"Im Prinzip müsste man wochenweise Deadlines als solche vorgeben und die Studis können sich nach irgendwelchen Regeln für jede Deadline Aufgaben zusammensuchen. D.h. sie müssten sich tatsächlich eine Art Konfiguration zusammen bauen (lassen) - vielleicht könnten wir dann eine Art GUI anbieten, wo man die Deadlines und Aufgaben auswählen (anklicken) kann, und wo dann das passende .yaml rausfällt? Die GUI selbst sollte auf einer .yaml von uns basieren, d.h. mit den Deadlines und Aufgaben und so und daraus dynamisch eine Ansicht generieren."

_Originally posted by @cagix in https://github.com/Programmiermethoden/2022-freiraum-dungeon/pull/146#discussion_r1221263741_

Edit: Die Frage ist, ob das **webbasiert** sein muss - das zieht gleich wieder Fragen nach einem Server o.ä. nach sich. Mir würde eine einfache Swing-basierte Lösung völlig reichen, die man per .jar verteilt und die die Studis direkt starten können.

**Comment by AKirsch1** [comment]
Meine Antwort aus dem PR auch hier:

Ich persönlich finde eine webbasierte Lösung immer schöner als Swing. Das sieht immer so alt aus. Es sollte aber natürlich eine statische Webseite sein, sonst stimme ich dir da mit dem Server definitiv zu. Auf der anderen Seite kenne ich aber auch nicht die Problematiken mit der PM-Lecture-Seite, die ich für vergleichbar komplex/simpel halte.

**Comment by cagix** [comment]
@AKirsch1 Ja, bin ich bei Dir. Aber wir haben keinen Server, und Github.io nutze ich nicht, weil die Frage nach dem korrekten Impressum und der Datenschutzerklärung ziemlich offen ist.

Im ILIAS gibt es eine Art eingebetteten Webserver aka "HTML-Lernmodul". Dort kann man eine statische Webseite reintun, das mache ich quasi mit meinem Vorlesungsmaterial. Wenn wir das Ding als statische Seite bauen können, dann könnten wir das auch direkt mit den VL-Unterlagen verbinden. Ansonsten halt der Swing- oder JavaFX-Weg.

**Comment by AKirsch1** [comment]
Dann haben wir mit dem HTML-Lernmodul ja schon unsere Lösung, falls wir mit dem webbasierten Weg gehen sollten.

**Comment by cagix** [comment]
> Dann haben wir mit dem HTML-Lernmodul ja schon unsere Lösung, falls wir mit dem webbasierten Weg gehen sollten.

ja genau. aber eben nur statische seiten.

**Comment by cagix** [comment]
@AKirsch1 evtl. thesis-label?

**Comment by AKirsch1** [comment]
> evtl. thesis-label?

Ist angeheftet. Klingt für mich zwar erstmal zu wenig für eine Bachelorarbeit, aber vielleicht findet man ja noch eine passende Ergänzung.

**Comment by cagix** [comment]
> > evtl. thesis-label?
> 
> 
> 
> Ist angeheftet. Klingt für mich zwar erstmal zu wenig für eine Bachelorarbeit, aber vielleicht findet man ja noch eine passende Ergänzung.

genau das war der gedanke dahinter. manche sachen sind zu groß, manche muss man vielleicht kombinieren, und manche werden vielleicht doch noch im projekt umgesetzt :)

edit: kam das nur mir grad so vor oder klang das fast so wie galadriels rede an ihrem mirror? 🤣
